core/src/main/java/com/graphhopper/util/details/IntersectionValues.java

```diff
@@ -0,0 +1,83 @@
+/*
+ *  Licensed to GraphHopper GmbH under one or more contributor
+ *  license agreements. See the NOTICE file distributed with this work for
+ *  additional information regarding copyright ownership.
+ *
+ *  GraphHopper GmbH licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except in
+ *  compliance with the License. You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package com.graphhopper.util.details;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class IntersectionValues {
+    public int bearing;
+    public boolean entry;
+    public boolean in;
+    public boolean out;
+
+    /**
+     *  create a List of IntersectionValues from a PathDetail
+     */
+    public static List<IntersectionValues> createList(Map<String, Object> intersectionMap) {
+        List<IntersectionValues> list = new ArrayList<>();
+
+        List<Integer> bearings = (List<Integer>) intersectionMap.get("bearings");
+        Integer in = (Integer) intersectionMap.get("in");
+        Integer out = (Integer) intersectionMap.get("out");
+        List<Boolean> entry = (List<Boolean>) intersectionMap.get("entries");
+
+        if (bearings.size() != entry.size()) {
```

Comment:
I am not sure about this case, so this is an error case, right? Should we return silent? Or should we at least log the error or even throw an IllegalStateException? This should never happen right?

Reply:
AFAIK this should never happen. Not sure what is an appropriate reaction on that. I wasn't aware that IllegalStateException is a RuntimeException, so we do not need to attribute each throw , which is good.
But I can create an Logger instance and Log it as well... Please advise me here.

Reply:
Yes, this should result in a HTTP request with response code 400, as Dropwizard catches this. Since we believe this should never happen, I'd say let's throw an exception at least (and maybe even log the result).